plugin/loader/initializer.go

```diff
@@ -20,24 +20,37 @@ func initialize(plugins []plugin.Plugin) error {
 
 func run(plugins []plugin.Plugin) error {
 	for _, pl := range plugins {
-		err := runIPLDPlugin(pl)
-		if err != nil {
-			return err
+		switch pl.(type) {
+		case plugin.PluginIPLD:
+			err := runIPLDPlugin(pl.(plugin.PluginIPLD))
+			if err != nil {
+				return err
+			}
+		case plugin.PluginTracer:
+			err := runTracerPlugin(pl.(plugin.PluginTracer))
+			if err != nil {
+				return err
+			}
+		default:
+			panic(pl)
 		}
 	}
 	return nil
 }
 
-func runIPLDPlugin(pl plugin.Plugin) error {
-	ipldpl, ok := pl.(plugin.PluginIPLD)
-	if !ok {
-		return nil
+func runIPLDPlugin(pl plugin.PluginIPLD) error {
+	err := pl.RegisterBlockDecoders(format.DefaultBlockDecoder)
+	if err != nil {
+		return err
 	}
 
-	err := ipldpl.RegisterBlockDecoders(format.DefaultBlockDecoder)
+	return pl.RegisterInputEncParsers(coredag.DefaultInputEncParsers)
+}
+
+func runTracerPlugin(pl plugin.PluginTracer) error {
+	err := pl.InitGlobalTracer()
```

Comment:
To make type checks easier, I would do:
```go
tr, err := pl.GetTracer()
if err != nil {
  return err
}
opentrace.SetGlobalTracer(tr)
```

That way if the types in your plugin are wrong, this won't fail silently, it will hit type assertion issues and throw an error.